Remove the "MUST" wording which requires exemplar to be on-by-default (#2414)
open-telemetry/opentelemetry-specification: specification/metrics/sdk.md

A Metric SDK MUST provide a mechanism to sample `Exemplar`s from measurements
via the `ExemplarFilter` and `ExemplarReservoir` hooks.

> `Exemplar` sampling SHOULD be turned off by default. If `Exemplar` sampling is

Reviewer: ```suggestion
`Exemplar` sampling MAY be enabled by default. If `Exemplar` sampling is
```

Author: I'm fine with either way, let's wait and see if other reviewers have strong opinion.

@yurishkuro what do you think?

Reviewer 2: Not a strong opinion, but I refer "should be off" because it gives a clear guideline to the SDKs for consistent behavior, instead of encouraging each SDK to pick their own default.
